File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        int n=A.size();
        vector<int>ans(n);
        unordered_map<int,int>mpp;
        for(int i=0;i<n;i++){
            mpp[A[i]]++;
            mpp[B[i]]--;
            int common=0;
            for(auto it:mpp){
                if(it.second==0) common++;
            }
            ans[i]=common;
        }
        return ans;
    }
};

File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.java
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        int n=A.length;
        int ans[]=new int[n];
        Map<Integer,Integer>mpp=new HashMap<>();
        for(int i=0;i<n;i++){
            mpp.put(A[i],mpp.getOrDefault(A[i],0)+1);
            mpp.put(B[i],mpp.getOrDefault(B[i],0)-1);
            int common=0;
            for(int x:mpp.keySet()){
                if(mpp.get(x)==0) common++;
            }
            ans[i]=common;
        }
        return ans;
    }
}